protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+	rest_client: Client,
+	faucet_client: FaucetClient,
+	///The signer account
+	signer: LocalAccount,
+}
+
+impl MovementClient {
+	pub async fn build_with_config() -> Result<Self, anyhow::Error> {
+		let dot_movement = dot_movement::DotMovement::try_from_env().unwrap();
+		let suzuka_config =
+			dot_movement.try_get_config_from_json::<suzuka_config::Config>().unwrap();
+		let node_connection_address = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_rest_connection_hostname;
+		let node_connection_port =
+			suzuka_config.execution_config.maptos_config.client.maptos_rest_connection_port;
+
+		let node_connection_url =
+			format!("http://{}:{}", node_connection_address, node_connection_port);
+		let node_connection_url = Url::from_str(node_connection_url.as_str()).unwrap();
+
+		let faucet_listen_address = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_faucet_rest_connection_hostname;
+		let faucet_listen_port = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_faucet_rest_connection_port;
+		let faucet_url = format!("http://{}:{}", faucet_listen_address, faucet_listen_port);
+		let faucet_url = Url::from_str(faucet_url.as_str()).unwrap();
+
+		let rest_client = Client::new(node_connection_url.clone());
+		let faucet_client = FaucetClient::new(faucet_url, node_connection_url.clone());
+
+		let seed = [3u8; 32];
+		let mut rng = rand::rngs::StdRng::from_seed(seed);
+		let signer = LocalAccount::generate(&mut rng);
+
+		Ok(MovementClient {
+			initiator_address: Vec::new(), //dummy for now
+			rest_client,
+			faucet_client,
+			counterparty_address: DUMMY_ADDRESS,
+			signer,
+		})
+	}
+}
+
+impl Clone for MovementClient {
+	fn clone(&self) -> Self {
+		todo!()
+	}
+}
+
+#[async_trait::async_trait]
+impl BridgeContractCounterparty for MovementClient {
+	type Address = MovementAddress;
+	type Hash = [u8; 32];
+
+	async fn lock_bridge_transfer_assets(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+		hash_lock: HashLock<Self::Hash>,
+		time_lock: TimeLock,
+		initiator: InitiatorAddress<Vec<u8>>,
+		recipient: RecipientAddress<Self::Address>,
+		amount: Amount,
+	) -> BridgeContractCounterpartyResult<()> {
+		//@TODO properly return an error instead of unwrapping
+		let args = vec![
+			self.to_bcs_bytes(&initiator.0).unwrap(),
+			self.to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
+			self.to_bcs_bytes(&hash_lock.0).unwrap(),
+			self.to_bcs_bytes(&time_lock.0).unwrap(),
+			self.to_bcs_bytes(&recipient.0).unwrap(),
+			self.to_bcs_bytes(&amount.0).unwrap(),
+		];
+		let payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"lock_bridge_transfer_assets",
+			self.counterparty_type_args(Call::Lock),
+			args,
+		);
+		let _ = utils::send_aptos_transaction(&self.rest_client, &mut self.signer, payload)
+			.await
+			.map_err(|_| BridgeContractCounterpartyError::LockTransferAssetsError);
```

Comment:
You are mapping an error but neither using it nor effectively short-circuiting it. So why are you doing this? It has no value.